lib/rex/proto/dns/resolver.rb

```diff
@@ -115,6 +116,8 @@ def proxies=(prox, timeout_added = 250)
     # @param argument
     # @param type [Fixnum] Type of record to look up
     # @param cls [Fixnum] Class of question to look up
+    #
+    # @return [Dnsruby::Message] DNS response
```

Comment:
Or Nil as well?

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/16390/commits/4a38939e5d02160716d855c18063e2a3e5ee5f72

Reply:
Technically your right in the event of an exception we wouldn't even hit that, we'd actually return whatever is in a handler:

```
irb(main):053:1* begin
irb(main):054:1*   raise NoMemoryError 
irb(main):055:1*   return 22 
irb(main):056:1* rescue NoMemoryError
irb(main):057:1*   print "A"
irb(main):058:1*   99
irb(main):059:0> end
A=> 99
irb(main):060:0> 
```

I'll just go ahead and remove that line since I don't think it serves any purpose since execution won't continue past the `raise` statement.

Reply:
Fixed in f1c8b17da3b1fe26e71a28a53a89604dc436123f